ihmc-communication/src/test/java/us/ihmc/communication/ROS2ToolsTest.java

```diff
@@ -162,4 +169,74 @@ public void testLongStringCommunication()
 
       ThreadTools.sleepForever();
    }
+
+   @Test
+   public void testLoopbackROS2Node() throws InterruptedException, IOException
+   {
+      AtomicBoolean messageReceived = new AtomicBoolean(false);
+      AtomicBoolean failed = new AtomicBoolean(false);
+      StringBuilder stringToSend = new StringBuilder("Hello World!");
+
+      ROS2Node loopbackNode = ROS2Tools.createLoopbackROS2Node(PubSubImplementation.FAST_RTPS, getClass().getSimpleName() + "LoopbackNode");
+      ROS2Subscription<String> loopbackSubscriber = loopbackNode.createSubscription(ROS2Tools.IHMC_ROOT.withType(String.class), subscriber ->
+      {
+         String message = subscriber.takeNextData();
+         messageReceived.set(true);
+         synchronized (messageReceived)
+         {
+            messageReceived.notify();
+         }
+         assertEquals(stringToSend.toString(), message.getDataAsString());
+         if (!stringToSend.toString().equals(message.getDataAsString()))
+            failed.set(true);
+      });
+
+      InetAddress outsiderAddress = getOutputAddress();
+      LogTools.info("Outsider node on {}", outsiderAddress);
+      ROS2Node outsiderNode = new ROS2Node(PubSubImplementation.FAST_RTPS,
+                                           getClass().getSimpleName(),
+                                           new RTPSCommunicationFactory().getDomainId(),
+                                           getOutputAddress());
+      ROS2Subscription<String> outsideSubscriber = outsiderNode.createSubscription(ROS2Tools.IHMC_ROOT.withType(String.class), subscriber ->
+      {
+         LogTools.error("Outsider node should NOT receive any messages");
+         failed.set(true);
+      });
+
+      ROS2PublisherBasics<String> stringPublisher = loopbackNode.createPublisher(ROS2Tools.IHMC_ROOT.withType(String.class));
+      String messageToSend = new String();
+      messageToSend.setData(stringToSend.toString());
+      stringPublisher.publish(messageToSend);
+
+      synchronized (messageReceived)
+      {
+         messageReceived.wait(1000);
+      }
+      assertTrue(messageReceived.get());
+      assertFalse(failed.get());
+
+      loopbackSubscriber.remove();
+      outsideSubscriber.remove();
+   }
+
+   private InetAddress getOutputAddress() throws IOException
+   {
+      Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+      while (interfaces.hasMoreElements())
+      {
+         NetworkInterface intrface = interfaces.nextElement();
```

Comment:
`intrface` typo

Reply:
I did this on purpose because `interface` is a java key word. Could do `networkInterface` instead.

Reply:
oh, duh. I think I usually use `iface`, you can just use intrface though